src/components/app.st.css

```diff
@@ -41,6 +45,7 @@
 }
 
 .root {
+    -st-states: hasError;
```

Comment:
if you read this comment first then ignore it. come back to this line after you read the entire review.